### PR TITLE
bn: introduce .toArrayLike() to speedup .toBuffer()

### PR DESCRIPTION
**Pull request by axic**
As discussed in https://github.com/indutny/bn.js/pull/76

### CODE REVIEW

**Comment by fanatid** [comment]
:+1:

File: lib/bn.js [hunk 1]
```diff
@@ -514,7 +518,7 @@
 
     this.strip();
     var littleEndian = endian === 'le';
-    var res = new Array(reqLength);
+    var res = new Cons(reqLength);
```
Comment (dcousens) [inline lib/bn.js]: `s/Cons/ArrayType`?

**Comment by dcousens** [comment]
@indutny LGTM :)

File: lib/bn.js [hunk 2]
```diff
@@ -503,10 +503,14 @@
...
     assert(typeof Buffer !== 'undefined');
-    return new Buffer(this.toArray(endian, length));
+    return this.toArrayLike(Buffer, endian, length);
   };
 
   BN.prototype.toArray = function toArray (endian, length) {
+    return this.toArrayLike(Array, endian, length);
+  };
+
+  BN.prototype.toArrayLike = function toArray (Cons, endian, length) {
```
Comment (indutny) [inline lib/bn.js]: `toArrayLike` in function's name ;) . Is there any point in making it public? If you think so - please add docs to readme

Comment (dcousens) [inline lib/bn.js]: @indutny I guess you could do `toArrayLike(Int8Array, ...)`

**Comment by axic** [comment]
All changes done

**Comment by dcousens** [comment]
re-ACK

**Comment by indutny** [comment]
Landed and released, thank you!

**Pull request closed without merging**